build.py

```diff
@@ -210,6 +210,10 @@ def gen_core(self):
         ("warning_level", "DEFAULT"),
       ]
 
+    # Add Closure provide, require and inherits methods.
+    params.append(("js_code", """var goog=goog||{};goog.provide=function(){};goog.require=function(){};
+goog.inherits=function(a,b){function c(){}c.prototype=b.prototype;a.superClass_=b.prototype;a.prototype=new c;a.prototype.constructor=a;};"""))
```

Comment:
Neat idea, but this moves some of our core code into a very unexpected location, build.py.  And it means everyone's build process needs a similar hack added.

Reply:
I'm happy moving these to a `base.min.js` file or something so they're on their own and under `closure/goog/`, but I can't think of a way to have them defined in `core/` and not have to:
- reimplement provide/require (as we need it in there for the reasons outlined in the PR description)
- have a need to depend (ie: require) on a "base" file anywhere we do a `goog.inherit`.
If we think of this patch as a way to make our code work with the closure compiler, then the build script is the appropriate place to patch them in.
For example, we have similar code [here](https://github.com/google/blockly/blob/master/build.py#L255) and [here](https://github.com/google/blockly/blob/master/build.py#L283).

Reply:
Going to go ahead with Blockly.utils.object.inherits....